packages/select/src/interfaces.ts

```diff
@@ -0,0 +1,119 @@
+} from '@xcritical/theme';
+import { IndicatorProps } from 'react-select/src/components/indicators';
+
+
+export type DropdownIndicatorProps = IndicatorProps<IOptionItem> & {
+  getStyles: any;
+};
+
+export type ClearIndicatorProps = IndicatorProps<IOptionItem> & {
+  getStyles: any;
+};
+
+export type MultiValueRemoveProps = IndicatorProps<IOptionItem> & {
+  getStyles: any;
+};
+
+export interface IIsRTL {
+  isRTL?: boolean;
+}
+
+export interface IThemeProps {
+  theme: IThemeNamespace<SelectTheme>;
+  appearance: string;
+  baseAppearance: string;
+}
+
+export interface ISelectOnlyProps {
+
+  appearance?: string;
+  baseAppearance?: string;
+  disabled?: boolean;
+  isMulti?: boolean;
+  isSearchable?: boolean;
+  isOpenMenu?: boolean;
+  items?: { [key: string]: IOptionItem };
+  value?: string | { [key: string]: IOptionItem };
+  className?: string;
+  shouldFitContainer?: boolean;
+  isRTL?: boolean;
+  isCloseMenuOnSelect?: boolean;
+  isHideSelectedOptions?: boolean;
+  isControlShouldRenderValue?: boolean;
+  textPosition?: string;
+  placeholder?: string;
+  theme?: IThemeNamespace<SelectTheme>;
+}
+
+export type SelectProps = Omit<Props<IOptionProps>, 'theme' | 'value'> & ISelectOnlyProps;
+
+export interface ISubComponents {
+  dropdown?: IThemeBase<ISelectCssTheme>;
+  dropdownList?: IThemeBase<ISelectCssTheme>;
+  button?: IThemeBase<ISelectCssTheme>;
+  indicatorsContainer?: IThemeBase<ISelectCssTheme>;
+  indicatorSeparator?: IThemeBase<ISelectCssTheme>;
+  dropdownIndicator?: IThemeBase<ISelectCssTheme>;
+  clearIndicator?: IThemeBase<ISelectCssTheme>;
+  loadingIndicator?: IThemeBase<ISelectCssTheme>;
+  loadingMessage?: IThemeBase<ISelectCssTheme>;
+  input?: IThemeBase<ISelectCssTheme>;
+  placeholder?: IThemeBase<ISelectCssTheme>;
+  labelText?: IThemeBase<ISelectCssTheme>;
+  option?: IThemeBase<ISelectCssTheme>;
+  valueContainer?: IThemeBase<ISelectCssTheme>;
+  singleValue?: IThemeBase<ISelectCssTheme>;
+  multiValue?: IThemeBase<ISelectCssTheme>;
+  multiValueLabel?: IThemeBase<ISelectCssTheme>;
+  multiValueRemove?: IThemeBase<ISelectCssTheme>;
+  noOptionsMessage?: IThemeBase<ISelectCssTheme>;
+}
+
+export interface ISelectCssTheme extends IStylesBase {
+  divided?: {
+    color: string;
+  };
+  boxShadow?: string;
+  prefixSpacing?: number;
+  postfixSpacing?: number;
+}
+
+export interface ILabelTextCssTheme extends ISelectCssTheme {
+  prefixSpacing?: number;
+  postfixSpacing?: number;
+}
+
+export type ISelectTheme = ISelectCssTheme & ISubComponents;
+
+export interface IReturnFunction< TValue> {
+  (theme: IThemeNamespace<SelectTheme>, appearance: string, baseAppearance: string): TValue;
+}
+
+export interface IReturnWithArgsFunction<TProp, TValue> {
+  (elementName: string, ...props: TProp[]): TValue;
+}
+
+export type GetStyles<TResult = { [key: string]: any }> =
+  IReturnFunction<IReturnWithArgsFunction<any, TResult>>;
+
+export type SelectTheme = ITheme<ISelectTheme>;
```

Comment:
Look like it also should be interface, but it will cause name duplication. What do you think about it?